File: ESTRUTURAS/1162/main.js
function leastSwaps(l, carriages) {
    let swaps = 0;
    for (let i = 0; i < l - 1; i++) {
        for (let j = i + 1; j < l; j++) {
            swaps += carriages[i] > carriages[j];
        }
    }
    return swaps;
}

let lines = require('fs').readFileSync('/dev/stdin', 'utf-8').split('\n');
let n = parseInt(lines.shift());
while (n--) {
    let l = parseInt(lines.shift());
    let carriages = lines.shift().split(' ').map(Number);
    console.log(`Optimal train swapping takes ${leastSwaps(l, carriages)} swaps.`);
}

File: ESTRUTURAS/1162/main.dart
import 'dart:io';

int leastSwaps(int l, List<int> carriages) {
  int swaps = 0;
  for (int i = 0; i < l - 1; i++) {
    for (int j = i + 1; j < l; j++) {
      if (carriages[i] > carriages[j]) swaps++;
    }
  }
  return swaps;
}

void main() {
  int n = int.parse(stdin.readLineSync()!);
  while (n-- > 0) {
    int l = int.parse(stdin.readLineSync()!);
    List<int> carriages = stdin.readLineSync()!.split(' ').map(int.parse).toList();
    print('Optimal train swapping takes ${leastSwaps(l, carriages)} swaps.');
  }
}
